packages/wonder-blocks-modal/components/modal-panel.js

```diff
@@ -102,34 +134,13 @@ export default class ModalPanel extends React.Component<Props> {
             });
         }
 
-        // Figure out the background color that'll be behind the close button
-        // If a titlebar is specified then we use that, otherwise we use the
-        // default background color.
-        const topBackgroundColor = (titleBar && titleBar.props.color) || color;
-
         return (
             <View
                 style={[styles.wrapper, color === "dark" && styles.dark, style]}
             >
-                {showCloseButton && (
-                    <View
-                        style={[
-                            styles.closeButton,
-                            (mediaSize) =>
-                                mediaSize === "small" &&
-                                styles.smallCloseButton,
-                        ]}
-                    >
-                        <ModalCloseButton
-                            color={
-                                topBackgroundColor === "dark" ? "light" : "dark"
-                            }
-                            onClick={onClickCloseButton}
-                        />
-                    </View>
-                )}
                 {titleBar}
                 {mainContent}
+                {this.maybeRenderCloseButton()}
```

Comment:
Thanks for extracting that.  This is an interesting issue.  Even if we wrote a test with enzyme it wouldn't have picked this up b/c enzyme could've simulated an click event and sent it directly to the close button itself whereas a user would still be unable to click the button.

Reply:
Yeah, I don't know how we could automate a test for something like this without a proper native test app (like something that runs natively and manipulates the OS pointer directly to simulate a real user).